src/login/logind-dbus.c

```diff
@@ -1922,6 +1922,10 @@ static int method_do_shutdown_or_sleep(
 
         (void) setup_wall_message_timer(m, message);
 
+        if (IN_SET(a->handle, HANDLE_HALT, HANDLE_REBOOT, HANDLE_POWEROFF, HANDLE_KEXEC, 
```

Comment:
`HANDLE_SOFT_REBOOT` is missing

Reply:
please use th ehelpers in #29377.

But yeah, an assert should be more appropriate here